File: app/src/main/java/idrisadetunmbi/githubusers/services/GithubService.java
package idrisadetunmbi.githubusers.services;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class GithubService {

    private Retrofit mRetrofit = null;

    public GithubApi getAPI() {
        final String BASE_URL = "https://api.github.com/";
        if (mRetrofit == null) {
            mRetrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return mRetrofit.create(GithubApi.class);
    }
}

File: app/src/main/java/idrisadetunmbi/githubusers/services/GithubService.kt
package idrisadetunmbi.githubusers.services

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class GithubService {

    private var mRetrofit: Retrofit? = null

    val api: GithubApi
        get() {
            val BASE_URL = "https://api.github.com/"
            if (mRetrofit == null) {
                mRetrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return mRetrofit!!.create(GithubApi::class.java)
        }
}
